languagetool-language-modules/de/src/main/java/org/languagetool/rules/de/GermanSpellerRule.java

```diff
@@ -2270,83 +2311,161 @@ protected boolean ignorePotentiallyMisspelledWord(String word) throws IOExceptio
+
+  private boolean isValidWordLength(String word) {
+    return word.length() <= MIN_WORD_LENGTH || word.length() >= MAX_WORD_LENGTH;
+  }
+
+  private boolean isValidPartLength(List<String> parts) {
     // don't assume very short parts (like "Ei") are correct, these can easily be typos:
-    if ((hasInfixS || part1.endsWith("s")) && part1.length() >= 4 /* includes 's' */ && part2.length() >= 3 && startsWithLowercase(part2)) {
-      String part1noInfix = part1.substring(0, part1.length()-1);
-      String part2uc = uppercaseFirstChar(part2);
-      if ((compoundPatternWithHeit.matcher(part1).matches() || wordsNeedingInfixS.contains(part1noInfix)) &&
-          isNoun(part2uc)) {
-        if (compoundPatternWithAction.matcher(part1noInfix).matches() ||
-            compoundPatternWithFirst.matcher(part2uc).matches() ||
-            part1.endsWith("schwungs") || part1.endsWith("sprungs") || isMisspelled(part1noInfix) || isMisspelled(part2uc)) {
-          return false;
+    if (parts.size() == 2) {
+      return parts.get(0).length() >= 3 && parts.get(1).length() >= 4;
+    }
+    if (parts.size() == 3) {
+      return parts.get(0).length() >= 3 && parts.get(1).length() >= 4 && parts.get(2).length() >= 4;
+    }
+    return false;
+  }
+
+  private boolean isOldSpelling(List<String> parts) {
+    for (String part: parts) {
+      String cleaned_part = removeTrailingSAndHyphen(part);
+      if (part.endsWith("s")) {
+        if (oldSpelling.contains(uppercaseFirstChar(part)) || oldSpelling.contains(uppercaseFirstChar(cleaned_part))
+          || oldSpelling.contains(lowercaseFirstChar(part)) ||oldSpelling.contains(lowercaseFirstChar(cleaned_part))) {
+          return true;
+        }
+      } else {
+        if (oldSpelling.contains(uppercaseFirstChar(cleaned_part)) || oldSpelling.contains(lowercaseFirstChar(cleaned_part))) {
+          return true;
         }
-        return true;
       }
     }
-    String part2uc = uppercaseFirstChar(part2);
-    if (!hasInfixS &&
-        part1.length() >= 3 && part2.length() >= 4 &&
-        !part2.contains("-") &&
-        startsWithLowercase(part2) &&
-        !part1.equals("Lass") &&  // e.g. "Lasstest" - couldn't find a more generic solution yet
-        (wordsWithoutInfixS.contains(part1) || (compoundPatternSpecialEnding.matcher(part1).matches() && isNoun(part2uc))) &&
-        (!isMisspelled(part1) || germanPrefixes.contains(lowercaseFirstChar(part1))) &&
-        !isMisspelled(part2uc) &&  // needed to not accept e.g. Alkoholgenuß ("Genuß" is accepted as a noun above)
-        isNoun(part2uc) // don't accept e.g. "Azubikommt"
-      ) {
-      //System.out.println("compound: " + part1 + " " + part2 + " (" + word + ")");
-      return true;
-    }
     return false;
   }
 
+  private boolean isValidGenderNeutralWord(List<String> parts, String word) {
+    int start = 0;
+    for (String part : parts) {
+      int end = start + part.length();
+
+      String to_check = word.substring(start, end);
```

Comment:
please use CamelCase (`toCheck`)

Reply:
I don't think I can just check for CamelCase here. Since the Binnen-I can be used for gendering, I need to check the individual parts of the compound as well ('SEO-ExpertInnen' VS 'SEO-ExpertInnern')

Reply:
I meant the variable name, it should be `toCheck` instead of `to_check`, according to Java conventions.